src/tests/unittests/asn1.c

```diff
@@ -114,42 +131,80 @@ static void torture_integer(void **state)
 	u8 min4[] = {0x80, 0x00, 0x00, 0x00};
 	u8 over[] = {0x7F, 0xFF, 0xFF, 0xFF, 0xFF};
 	int value;
+	u8 *buf = NULL;
+	size_t buflen = 0;
 	int rv = 0;
 
 	rv = sc_asn1_decode_integer(zero, sizeof(zero), &value);
 	assert_int_equal(rv, SC_SUCCESS);
 	assert_int_equal(value, 0);
+	rv = asn1_encode_integer(value, &buf, &buflen);
+	assert_int_equal(rv, SC_SUCCESS);
+	assert_int_equal(buflen, sizeof(zero));
+	assert_memory_equal(buf, zero, buflen);
+	free(buf);
 
 	rv = sc_asn1_decode_integer(one, sizeof(one), &value);
 	assert_int_equal(rv, SC_SUCCESS);
 	assert_int_equal(value, 1);
+	rv = asn1_encode_integer(value, &buf, &buflen);
+	assert_int_equal(rv, SC_SUCCESS);
+	assert_int_equal(buflen, sizeof(one));
+	assert_memory_equal(buf, one, buflen);
+	free(buf);
```

Comment:
test cases like this really are asking for parametrisation, there is a lot of code repetition here

Reply:
so a macro would still work, wouldn't it?